### PR TITLE
Add implementation for builtin function 'period_add'

### PR DESCRIPTION
**Pull request by myrice**
xi9 0 _stj ev r6i1 yq0

### CODE REVIEW

File: expression/builtin_time.go [hunk 1]
```diff
@@ -1835,7 +1835,55 @@ type builtinPeriodAddSig struct {
...
+	months, err := args[1].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	sum := time.Month(m + months)
+	// TODO: Consider time_zone variable.
+	t := time.Date(int(y), sum, 1, 0, 0, 0, 0, time.Local)
+
+	//ret := fmt.Sprintf("%d%d", t.Year(), t.Month())
```
Comment (hanfei1991) [inline expression/builtin_time.go]: remove this

File: expression/builtin_time.go [hunk 2]
```diff
@@ -1835,7 +1835,55 @@ type builtinPeriodAddSig struct {
...
+	m := period % 100
+
+	// YYMM, 00-69 year: 2000-2069
+	// YYMM, 70-99 year: 1970-1999
+	// YYYYMM, 1000-? year: 1000-?
+	if y <= 69 {
+		y += 2000
+	} else if y <= 99 {
+		y += 1900
+	} else if y < 1000 {
```
Comment (hanfei1991) [inline expression/builtin_time.go]: y < 1000 is also valid 
mysql> select period_add(12323,10);
+----------------------+
| period_add(12323,10) |
+----------------------+
|                12509 |
+----------------------+
1 row in set (0.00 sec)

File: expression/builtin_time.go [hunk 3]
```diff
@@ -1835,7 +1835,55 @@ type builtinPeriodAddSig struct {
...
+	}
+
+	sc := b.ctx.GetSessionVars().StmtCtx
+	period, err := args[0].ToInt64(sc)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
+	//Check zero
+	if period <= 0 {
```
Comment (hanfei1991) [inline expression/builtin_time.go]: mysql> select period_add(0,1);
+-----------------+
| period_add(0,1) |
+-----------------+
|               0 |
+-----------------+
1 row in set (0.00 sec)

0 always return 0.

**Comment by hanfei1991** [comment]
@tiancaiamao PTAL

**Comment by hanfei1991** [comment]
@myrice resolve conflict please

**Comment by hanfei1991** [comment]
LGTM @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM
please resolve conflicts. @myrice

**Comment by tiancaiamao** [comment]
Seems gofmt error, run `make check` locally. @myrice

**Comment by tiancaiamao** [comment]
Conflict again...

**Comment by myrice** [comment]
Conflict has been resolved

**Comment by tiancaiamao** [comment]
Thanks for your contribution! @myrice

**Comment by myrice** [comment]
This experience is exciting! Thank you!@tiancaiamao @hanfei1991

**Pull request merged**